.circleci/config.yml

```diff
@@ -61,6 +62,20 @@ workflows:
           filters:
             tags:
               only: /.*/
+      - test-render-linux-firefox:
+          requires:
+            - prepare
+          filters:
+            tags:
+              only: /.*/
+      - test-render-macos-chrome:
+          filters:
```

Comment:
```suggestion
      - test-render-macos-chrome:
          requires:
            - prepare
          filters:
```

So that these build don't appear detached and wouldn't run if prepare fails:
<img width="321" alt="Screen Shot 2022-11-04 at 9 24 12 AM" src="https://user-images.githubusercontent.com/7061573/200026261-b4187ccb-18a6-4e04-b837-fbe2bfa977be.png">

Reply:
I made this change, but looking again at the content of `prepare`, I'm not sure if this makes much sense. My understanding of this job is that it's restoring the cache (effectively yarn install) This happens on the main linux "machine" (docker) so isn't helpful to the runs that depend on virtual machines.

Is this still valuable just to prevent them running if `prepare` fails?

Should we consider having a have `prepare-windows` and `prepare-macos` scripts? Using caching here would be an optimization, but setup is a relatively small part of the whole test run.

Reply:
The important part is that dependencies are correctly adjusted, as long as they are I think we're ok here. For example, we don't want a publishing/deployment of release running if those builds aren't satisfied. And at the moment they appear disconnected both in the input and output.

Here's an example of the release build from this morning (where deployment depends on previous test builds): https://app.circleci.com/pipelines/github/mapbox/mapbox-gl-js/12032/workflows/bfa7bf79-48f3-433e-a37c-2fb47f161bcc

<img width="755" alt="Screen Shot 2022-11-07 at 4 59 50 PM" src="https://user-images.githubusercontent.com/7061573/200448150-e0593168-0ebe-4994-a5e1-19455419e668.png">

Reply:
> For example, we don't want a publishing/deployment of release running if those builds aren't satisfied. And at the moment they appear disconnected both in the input and output.

I removed the virtual machine tests depending on `prepare` and added all three new tests as dependencies to both `deploy` scripts.